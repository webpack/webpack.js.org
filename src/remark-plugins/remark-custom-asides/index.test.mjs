import remark from 'remark';
describe('customize blockquote', () => {
  it('should transform W> into aside of warning', () => {
    remark()
      .use(require('./index.js'), {
        mapping: {
          'W>': 'warning',
        },
      })
      .use(require('remark-html'))
      .process(
        `
W> hello world
`,
        function (err, { contents }) {
          expect(err).toBeNull();
          expect(contents).toContain('<aside class="warning"');
          expect(contents).toContain('<h6 class="warning__prefix"');
          expect(contents).toContain('warning');
          expect(contents).toMatchSnapshot();
        }
      );
  });
});
