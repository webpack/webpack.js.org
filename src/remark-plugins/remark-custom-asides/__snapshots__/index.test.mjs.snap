// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`customize blockquote should transform W> into aside of warning 1`] = `
"<aside class=\\"warning\\"><h6 class=\\"warning__prefix\\">warning</h6><p>hello world</p></aside>
"
`;
